src/test/java/org/takes/rs/RsPrintTest.java

```diff
@@ -190,4 +218,52 @@ public boolean haveFlushed() {
             return this.flushed;
         }
     }
+
+    /**
+     * Fake wrapper for InputStream to make sure body is closed.
+     *
+     * @author Alena Gerasimova (olena.gerasimova@gmail.com)
+     * @version $Id$
+     * @since 2.0
+     */
+    private static final class FakeInput extends InputStream {
+
+        /**
+         * Have input been closed?
+         */
+        private boolean closed;
+
+        /**
+         * Origin.
+         */
+        private final InputStream origin;
+
+        /**
+         * Ctor.
+         * @param origin Origin input
+         */
+        FakeInput(final InputStream origin) {
+            super();
+            this.origin = origin;
+        }
+
+        @Override
+        public int read() throws IOException {
+            return this.origin.read();
+        }
+
+        @Override
+        public void close() throws IOException {
+            this.origin.close();
+            this.closed = true;
+        }
+
+        /**
+         * Have input been closed?
+         * @return True, if input wes closed, false - otherwise
+         */
+        public boolean haveClosed() {
```

Comment:
@olenagerasimova it would be more conventional to call method in present simple: `haveClosed` -> `isClosed`, because it describes current object state.